lib/puppet/util/at_fork.rb

```diff
@@ -0,0 +1,35 @@
+require 'puppet'
+
+# A module for building AtFork handlers. These handlers are objects providing
+# pre/post fork callbacks modeled after those registered by the `pthread_atfork`
+# function.
+# Currently there are two AtFork handler implementations:
+# - a noop implementation used on all platforms except Solaris (and possibly
+#   even there as a fallback)
+# - a Solaris implementation which ensures the forked process runs in a different
+#   contract than the parent process. This is necessary for agent runs stared by
+#   the puppet agent service to be able to restart that service without being
+#   killed in the process as a consequence of running in the same contract as the
+#   service.
+module Puppet::Util::AtFork
+  @handler_class = loop do
```

Comment:
Yes, exactly so, I wish there was a method similar to `loop` but which would have called the block only once.
I realize this is not a common use of the `loop` method that's why I added the comments on the break statements used to return values from the construct.